framework/auth/views.py

```diff
@@ -257,6 +301,62 @@ def confirm_email_get(token, auth=None, **kwargs):
     ))
 
 
+@must_be_logged_in
+def unconfirmed_email_remove(auth=None):
+    """Called at login if user cancels their merge or email add.
+    methods: DELETE
+    """
+    user = auth.user
+    json_body = request.get_json()
+    try:
+        given_token = json_body['token']
+    except KeyError:
+        raise HTTPError(http.BAD_REQUEST, data={
+            'message_short': 'Invalid token',
```

Comment:
These error messages are incorrect; should be "Missing token" and "Must provide a token".